### ISSUE TITLE
extend LazyFrame Attributes list

### ISSUE THREAD

**Issue by yuuuxt** [pola-rs/polars#3792]
#### Describe your feature request

Compared to `DataFrame`, `LazyFrame` doesn't have the following attributes: `shape`, `height`, `width`. 

It would be good to eliminate such a difference.

Is it reasonable & possible to have `df_lazy.shape` or like `df_lazy.shape.collect()`?

**Comment by ritchie46** [comment]
We could add the `width`. The `height` is unknown. We are talking about a promise of a computation, we don't know the result shape.

**Comment by yuuuxt** [comment]
Yes, I was only considering `LazyFrame` returned by `scan_parquet`, so was thinking `shape` is given.

**Comment by ritchie46** [comment]
> Yes, I was only considering `LazyFrame` returned by `scan_parquet`, so was thinking `shape` is given.

Even at that point we don't know the shape, as we don't read that

**Comment by thehappycheese** [comment]
I came here to suggest `lazyframe.shape` but [yuuuxt](https://github.com/yuuuxt) beat me to it :)

My expectation is that `df_lazy.shape.collect()` would either;

1. Return the shape lazily if available in the metadata (as is the case with parquet), or
2. Read the entire csv (or other formats) to get the shape

in my use case:

### Using eager method

```python
for item in files:
    print(ps.read_parquet(item).shape[0])
# takes 3.5 seconds
```

Pretty slow :(

### Using the best work-around I could think of

```python
import polars as ps
for item in files:
    df = ps.scan_parquet(item)
    print(ps.scan_parquet(item).select(ps.count(df.columns[0])).collect()[0,0])
# takes 1.9 seconds
```

Better, but not lazy enough!

### Using pyarrow to directly read metadata:

```python
from pyarrow.parquet import ParquetFile
def get_rows(file_name):
    pf = ParquetFile(file_name)
    return pf.metadata.num_rows
for item in files:
    print(get_rows(item))
# takes <0.1 seconds
```

Now that's more like what I expect.. near instantaneous!
